scripts/debugCycleRecords.ts

```diff
@@ -40,66 +42,139 @@ rl.on('line', (line) => {
 
 // When file reading is complete, log the completeCycles array or perform further processing
 rl.on('close', () => {
-  console.log('Finished reading the file. Processing complete.');
-  
-  const uniqueCompleteCycles = completeCycles.map((cycleArray) => {
-    const cycleMap = new Map();
+  const filteredCycles = completeCycles.map((cycle, index) => {
+    if (index < completeCycles.length - 1) {
+      const nextCyclePorts = new Set(completeCycles[index + 1].map((r) => r.port))
+      return cycle.filter((record) => nextCyclePorts.has(record.port))
+    }
+    return [] // we dont know if the last cycle will contain rotating out nodes so ignore it for now
```

Comment:
we actually got rid of the code that causes rotation OOS in the mega commit, so we can remove/comment out code that addresses that now